packages/qdrant-loader/src/qdrant_loader/core/pipeline/orchestrator.py

```
@@ -367,6 +378,57 @@ async def _detect_document_changes(
+
+        document_ids: list[str] = []
+        for document in deleted_documents:
+            try:
+                await self.components.state_manager.mark_document_deleted(
+                    document.source_type,
+                    document.source,
+                    document.id,
+                    project_id,
+                )
+                document_ids.append(document.id)
+                logger.debug(f"Marked document deleted in state: {document.id}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)